extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/dart/controller/sql/PrePlannedDartQueryMaker.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.msq.dart.controller.sql;
+
+import org.apache.druid.msq.exec.QueryKitBasedMSQPlanner;
+import org.apache.druid.msq.exec.ResultsContext;
+import org.apache.druid.msq.indexing.LegacyMSQSpec;
+import org.apache.druid.msq.indexing.QueryDefMSQSpec;
+import org.apache.druid.msq.kernel.QueryDefinition;
+import org.apache.druid.msq.logical.DruidLogicalToQueryDefinitionTranslator;
+import org.apache.druid.msq.sql.MSQTaskQueryMaker;
+import org.apache.druid.query.QueryContext;
+import org.apache.druid.query.QueryContexts;
+import org.apache.druid.server.QueryResponse;
+import org.apache.druid.server.security.ForbiddenException;
+import org.apache.druid.sql.calcite.planner.PlannerContext;
+import org.apache.druid.sql.calcite.rel.DruidQuery;
+import org.apache.druid.sql.calcite.rel.logical.DruidLogicalNode;
+import org.apache.druid.sql.calcite.run.QueryMaker;
+
+import java.util.List;
+import java.util.Map.Entry;
+
+/**
+ * Executes Dart queries with up-front planned {@link QueryDefinition}.
+ *
+ * Normal execution flow utilizes {@link QueryKit} to prdocue the plan;
+ * meanwhile it also supports planning the {@link QueryDefinition} directly from
+ * the {@link DruidLogicalNode}.
+ */
+class PrePlannedDartQueryMaker implements QueryMaker, QueryMaker.FromDruidLogical
+{
+  private PlannerContext plannerContext;
+  private DartQueryMaker dartQueryMaker;
+
+  public PrePlannedDartQueryMaker(PlannerContext plannerContext, QueryMaker queryMaker)
```

Comment:
```suggestion
  public PrePlannedDartQueryMaker(PlannerContext plannerContext, DartQueryMaker queryMaker)
```

Reply:
actually right now this makes more sense...originally it was more generic